content/blog/esp32-memory-map-101/index.md

```diff
@@ -0,0 +1,351 @@
+---
+title: "ESP32's family Memory Map 101"
+date: 2024-08-20T18:56:49+02:00
+tags: ["ESP32", "ESP32-S2", "ESP32-S3", "ESP32-C3", "ESP32-C6"]
+showAuthor: false
+authors:
+  - "marek-matej"
+---
+
+
+The performance and memory capabilities of the microcontrollers are increasing, and so is their complexity, especially when the users are responsible for mapping external memory chips such as Flash or SPIRAM by configuring a memory management unit.
+
+When I started to use Espressif ESP32’s series SoC’s with the [Zephyr RTOS](https://docs.zephyrproject.org/latest/index.html), quickly I realized that these SoC’s are a very different world, compared to my previous MCU experiences, which were mostly ARM-related. ARM Cortex-M’s - in most cases - uses Von-Neumann memory architecture, while the ESP32’s series of microcontrollers uses the Harvard memory architecture, which adds complexity.
```

Comment:
Regarding ESP32's series of SoC's:

First of all, the abbreviation SoCs is more common in Espressif documentation. To my surprise, Wikipedia, also [suggests](https://en.wikipedia.org/wiki/System_on_a_chip) this plural form.

There is a number of ways of how a subset of chips is referred to in Espressif documentation and articles: (1) ESP32 series of SoCs, (2) ESP32 family of SoCs, etc. Sometimes these terms are interchangeable. Sometimes, they are used side by side with modified meanings. At least in the hardware documentation, we started removing family wherever possible. Suggest that you also use ESP32 series of SoCs.

In summary, please consider using _ESP32 series of SoCs_.